clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -598,22 +608,22 @@ private void insertInSequenceOrder(Deque<ProducerBatch> deque, ProducerBatch bat
     /**
      * Add the leader to the ready nodes if the batch is ready
      *
-     * @param nowMs The current time
      * @param exhausted 'true' is the buffer pool is exhausted
      * @param part The partition
      * @param leader The leader for the partition
      * @param waitedTimeMs How long batch waited
      * @param backingOff Is backing off
+     * @param backoffAttempts Number of attempts for calculating backoff delay
      * @param full Is batch full
      * @param nextReadyCheckDelayMs The delay for next check
      * @param readyNodes The set of ready nodes (to be filled in)
      * @return The delay for next check
      */
-    private long batchReady(long nowMs, boolean exhausted, TopicPartition part, Node leader,
-                            long waitedTimeMs, boolean backingOff, boolean full,
-                            long nextReadyCheckDelayMs, Set<Node> readyNodes) {
+    private long batchReady(boolean exhausted, TopicPartition part, Node leader,
+                            long waitedTimeMs, boolean backingOff, int backoffAttempts,
+                            boolean full, long nextReadyCheckDelayMs, Set<Node> readyNodes) {
         if (!readyNodes.contains(leader) && !isMuted(part)) {
-            long timeToWaitMs = backingOff ? retryBackoffMs : lingerMs;
+            long timeToWaitMs = backingOff ? retryBackoff.backoff(backoffAttempts) : lingerMs;
```

Comment:
Here, backoffAttempts is already subtracted by one when calling retryBackoff.backoff. In other places, we pass in the true attempts and subtract by one when calling retryBackoff.backoff. It would be useful to make that consistent.